### PR TITLE
Prevents instant summons from recalling anchored objects

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Small bug fix that prevents anchored objects from being taken, without sacrificing the rest of its intended features.

## Why It's Good For The Game

Closes an issue from 2017 - https://github.com/tgstation/tgstation/issues/22837

## Changelog

:cl:
fix: Instant recall spell no longer recalls anchored objects, such as microwaves, if the marked item is placed inside of one.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
Gonna add an obj check when I can

**Comment by Fikou** [comment]
its hilarious as fuck to summon a deconstructive analyzer from rnd into your hand

**Comment by JohnFulpWillard** [comment]
im back to fix this and realized im a complete dumbfuck

**Comment by Shadowflame909** [comment]
Who stole the captains toilet??

**Comment by MetalClone** [comment]
Now I won't be able to summon and break the emergency survival pod shelter storages damn..

**Comment by anclandor** [comment]
Please don't remove this, I mean, yeah its a bug but it's also really funny to summon the deep fryer when then chef fries your weapon.

File: code/modules/spells/spell_types/summonitem.dm [hunk 1]
```diff
@@ -79,12 +78,15 @@
 						M.dropItemToGround(item_to_retrieve)
 
 					else
-						if(istype(item_to_retrieve.loc, /obj/machinery/portable_atmospherics/)) //Edge cases for moved machinery
-							var/obj/machinery/portable_atmospherics/P = item_to_retrieve.loc
+						var/obj/retrieved_item = item_to_retrieve.loc
+						if(retrieved_item.anchored)
+							break
```
Comment (optimumtact) [inline code/modules/spells/spell_types/summonitem.dm:83]: just return, don't break, the way they've done this of embedding this handling inside the for loop is unfortunate

Comment (JohnFulpWillard) [reply code/modules/spells/spell_types/summonitem.dm:83]: If it returns then it wouldn't get the item at all, I just didn't want it to take the whole machine its in, with it

Comment (optimumtact) [reply code/modules/spells/spell_types/summonitem.dm:83]: I'm not really sure what you mean by this, it only checks the loc once, there's no loop here?

**Comment by timothymtorres** [comment]
@JohnFulpWillard in the future can you please make sure you [link an issue to the PR](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) properly so that it autocloses when the PR gets merged.  This should help cleanout some old issues that were already fixed.

**Pull request merged**